app/components/buySell/BuySellDialog.js

```diff
@@ -0,0 +1,166 @@
+  dialogTitle: {
+    id: 'buysell.dialog.title',
+    defaultMessage: '!!!Buy / Sell ADA',
+  },
+  dialogSelectAddress: {
+    id: 'buysell.dialog.selectAddress',
+    defaultMessage: '!!!Please select the receiving address. This will be shared with the third party provider called Changelly for the buy / sell of ADA. ',
+  },
+  dialogDescription: {
+    id: 'buysell.dialog.instructions',
+    defaultMessage: '!!!Please select your preferences. On the next screen, confirm your selection by pressing the green arrow on the top right',
+  },
+  dialogManual: {
+    id: 'buysell.dialog.manual',
+    defaultMessage: '!!!I will add my address manually',
+  },
+});
+
+type WalletInfo = {|
+  walletName: string,
+  currencyName: string,
+  anAddressFormatted: string,
+|}
+
+type Props = {|
+  +onCancel: void => void,
+  +walletList: () => Promise<array<WalletInfo>>
+|};
+
+const WIDGET_URL = 'https://widget.changelly.com?from=*&to=*&amount=200&fromDefault=usd&toDefault=ada&theme=default&merchant_id=g9qheu8vschp16jj&payment_id=&v=3'
+
+type State = {|
+  addressSelected: ?string,
+  walletList: ?array<WalletInfo>,
+|};
+
+@observer
+export default class BuySellDialog extends Component<Props, State> {
+  static contextTypes: {|intl: $npm$ReactIntl$IntlFormat|} = {
+    intl: intlShape.isRequired,
+  };
+
+  state: State = {
+    addressSelected: null,
+    walletList: null,
+  };
+
+  createRows: ($npm$ReactIntl$IntlFormat, array<WalletInfo>) => Node = (intl, wallets) => (
+    wallets.map((wallet) => {
+      return (
+        <div className={styles.row}>
+          <div className={styles.left}>
+            <div className={styles.nameAndCurrency}>
+              { wallet.currencyName ? `(${wallet.currencyName}) ` : ''}{wallet.walletName}
+            </div>
+            <div className={styles.address}>
+              {truncateAddress(wallet.anAddressFormatted)}
+            </div>
+          </div>
+          <div className={styles.right}>
+            {/* Verify Address action */}
+            <button
+              type="button"
+              onClick={() =>
+                this.setState({ addressSelected: wallet.anAddressFormatted })
+              }
+            >
+              <div>
+                <span className={styles.verifyIcon}>
+                  <VerifyIcon />
+                </span>
+              </div>
+            </button>
+            {/* Action block end */}
+          </div>
+        </div>
+      )
+    })
+  )
+
+  render(): Node {
+    const { intl } = this.context;
+
+    if (this.state.walletList == null) {
+      this.props.walletList()
+        // eslint-disable-next-line promise/always-return
+        .then((resp) => {
+          const wallets = [
+            ...resp,
+            {
+              walletName: intl.formatMessage(messages.dialogManual),
+              currencyName: '',
+              anAddressFormatted: '',
+            }
+          ]
+          this.setState({ walletList: wallets })
```

Comment:
Feels strange we're updating the state based on the props in the render function -- even if it is in some other async context. 
We may want to use [a different lifecycle function](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) for this

Reply:
(conversation in Slack)